### DISCUSSION TITLE
Der Radiowecker auch als BT Lautsprecher

### DISCUSSION THREAD

**Discussion by CT7BHO** [beabel/radiowecker#11] [Ideen / Ideas]
Hallo,
der Radiowecker in seiner aktuellen Version 2.1.0 ist ja schon ein tolles Projekt.
Schön währe es, wenn er noch eine Funktion "BT-Lautsprecher" bekommen würde.
Ein Handy oder Tablet via Bluetooth mit dem Radiowecker verbinden und mp3 oder Höhrbuch wiedergeben.
Unser ESP32 kann ja nicht nur WLAN sondern auch Bluetooth. Aber geht auch beides zusammen?

**Comment by beabel** [comment]
Hi,
also ich habe eben ein wenig damit rumgebastelt.um jedoch die Bluetooth Funktionen mit einzuarbeiten habe ich mich für die Bibliothek von  [Phil Schatzmann](https://github.com/pschatzmann/ESP32-A2DP/tree/main) entschieden. Jedoch schon nur das simple includen der Bib übersteigt jetzt den Speicher.
Falls du jedoch andere Möglichkeiten kennst, gerne mehr Infos dazu.

**Reply by beabel** [reply]
_Der Sketch verwendet 1917973 Bytes (146%) des Programmspeicherplatzes._ und da habe ich nur die A2DP include, keinen weiteren code angepasst.

**Reply by CT7BHO** [reply]
Moin,
mit der A2DP habe ich auch schon experimentiert.
Mit einer der Beispiele aus der Bibliothek konnte ich auch schon einen reinen BT-LS ohne Schnick-Schnack realisieren.

In einem anderen Projekt, ein einfaches Internetradio, hatte ich dann beim einbinder der A2DP das gleicher Problem mit dem Speicher.

Im Arduino-Forum fand ich dann diesen Eintrag:

Die Arduino-IDE setzt eine willkürliche Grenze. Diese steht in der boards.txt (arduino/hardware/espressif/esp32). Dort steht z.B. für das ESP32 Dev Module und für alle anderen Boards:
esp32.upload.maximum_size=1310720.

Ändere bitte den Wert auf 3145728
Quelle: https://www.arduinoforum.de/arduino-Thread-ESP32-Programmspeicherplatz-zu-gering

Gesagt, getan: Mein ESP32 ging damit aber in eine Dauer-Reset-Schleife. Die Ursache habe ich noch nicht gefunden.
Meine Vermutung ist, das WLAN und BT hier einen konflickt haben. Beide verwenden ja das 2,4 GHz Modul.

Könnte man nicht auch via WLAN einen mp3-Stream an den ESP senden?

Grüße aus Portugal
Rainer

**Reply by beabel** [reply]
Nun Das Projekt soll jedoch ohne größere Eingriffe auskommen.
Das ist der Gedanke hinter den Projekten von AZ-Delivery.
Wenn ich jetzt noch erklären muss, das man txt Dateien anpassen muss etc. dann ist dieser Gedanke hinfällig.
Jedoch kann jeder gern anhand deines Anstoßes lokal experimentieren.
Ich werde jedoch das Projekt so belassen, das es mit einspielen sofort funktioniert.
Auch aus diesem Grund, um nicht von der Anleitung abzuweichen, habe ich die ArduinoJson als einzel .h Datei mit rein genommen.
Da sonst eine weitere Bibliothek eingebunden werden müßte, was nicht in der AZ Blog Anleitung drin steht.

**Reply by beabel** [reply]
Du könntest ja einen lokalen Netzwerk Stream statt eines Radio streams in die Radio stationsliste eintragen. Dazu brauchst du nat. Irgendeinen Dienst welcher dir einen Stream erstellt.

**Reply by CT7BHO** [reply]
Das ist eine tolle Idee, Das Höhrbuch im iPad via iTunes als Netzwerkstream wiedergeben und im Radiowecker die Station "iPad" wiedergeben. So, oder ähnlich. 
Mache mich gleich mal an Werk und werde berichten.

**Comment by Hotzi57** [comment]
Ich habe mir dafür in das Radio einen kleinen Bluetoothtransmitter für ca. 9€ eingebaut. Dazu noch ein kleiner Umschalter für Bluetoothbetrieb oder internen Lautsprecher. Fertig. Läuft super.

**Reply by CT7BHO** [reply]
Hallo Hotzi57, vielen Dank für Deinen Beitrag.
Wenn ich dich richtig verstehe, kannst du mit dem kleinen Umschalter festlegen, ob das Audiosignal vom Radiowecker über die eingebauten Lautsprecher oder über den BT-Transmitter wiedergegeben wird. Im BT-Betrieb hörst du dann das Radio mit einem BT-Kopfhörer oder BT-Lautsprecher.
Das ist eine interessante Variante und es stellt sich die Frage, ob man nicht auch via Software das Audiosignal über das ESP BT-Modul ausgeben kann. Also ohne HW-Erweiterung.
Mein wunsch ist es aber, das ich mein Mobiltelefon mit dem Radio koppeln kann um dann die Musikwiedergabe am Telefon zu starten und der Radiowecker gibt diese über seine eingebauten Lautsprecher wieder. Der Radiowecker wird also zum BT-Lautsprecher.
Es gibt Beispiele,, wo mit einem ESP32 ein BT-Lautsprecher realisiert wird. Habe ich auch schon getestet, funktioniert prima. Das Problem ist die gleichzeitige WLAN- und BT-Einbingung in einem Arduino-Projekt.
Grüße aus Portugal, Rainer.
